sympy/printing/fcode.py

```diff
@@ -178,6 +172,32 @@ def _print_Piecewise(self, expr):
                                       "inline operators is not supported in "
                                       "standards earlier than Fortran95.")
 
+    def _print_Assignment(self, expr):
```

Comment:
This is messy logic. What if the rhs contains sign as a subexpression? 

I take it Fortran doesn't allow assignments in expressions like C. 

We really need a good distinction (in all the code printers) between statements and expressions, and things that can only be printed as expressions (an expression being any piece of code that can be inserted arbitrarily in a larger expression, with the right parentheses).